### PR TITLE
Feature/com 17 create hardcoded array for triangle cube and other

### PR DESCRIPTION
**Pull request by Lukauigi**

### CODE REVIEW

File: Core/main.cpp [hunk 1]
```diff
@@ -3,6 +3,139 @@
...
+    GLCall(glGenVertexArrays(1, &vao));
+    GLCall(glBindVertexArray(vao));
+
+    // send vertex buffer to GPU
+    unsigned int vbo;
+    GLCall(glGenBuffers(1, &vbo));
+    GLCall(glBindBuffer(GL_ARRAY_BUFFER, vbo));
+    GLCall(glBufferData(GL_ARRAY_BUFFER, sizeof(positions), positions, GL_STATIC_DRAW));
+
+    std::string vertexShader =
```
Comment (le-rich) [inline Core/main.cpp:100]: Note for future: We sould probably find a way to write programs elsewhere and load them here

**Review by le-rich — DISMISSED** [review]
LGTM

- In future PRs please sign your name and the issue in square brackets for visibility.
[LB][COM-17]

**Review by Tissure — CHANGES_REQUESTED** [review]
Most of this code should be moved to a Rendering class. This keeps all the rendering code contained in one place and means we won't need to go back and refactor trying to extract all of the code out later.

File: Core/main.cpp [hunk 2]
```diff
@@ -3,6 +3,139 @@
...
+    GLCall(glLinkProgram(program));
+    GLCall(glValidateProgram(program));
+    // Delete the shaders since they have been linked to the program
+    GLCall(glDeleteShader(vs));
+    GLCall(glDeleteShader(fs));
+
+    return program;
+}
+
+void drawTriangle()
```
Comment (Tissure) [inline Core/main.cpp:81]: Additionally the feature is requesting for a hard coded cube and other primitives. These should get put into their own file like Primitives.cpp that can be sent to a generic draw function.

File: Core/primitives.h
Comment (le-rich) [inline Core/primitives.h:1]: This should not be in Core, please see Kyle's last PR to see structure

File: Core/renderer.cpp
Comment (le-rich) [inline Core/renderer.cpp:1]: This should not be in Core, please see kyle's last PR to see structure

File: Core/main.cpp [hunk 3]
```diff
@@ -1,8 +1,142 @@
...
 #include <GLFW/glfw3.h>
 
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <sstream>
+
+#include "primitives.h"
+
+#define ASSERT(x) if (!(x)) __debugbreak();
```
Comment (le-rich) [inline Core/main.cpp:11]: All these OpenGL functions are duplicated code from renderer.cpp, no duplicated code in repo please.

This should also not be in main.cpp, see Architecture's latest expectcations for structure

**Review by le-rich — CHANGES_REQUESTED** [review]

File: Core/renderer.cpp
Comment (Lukauigi) [reply Core/renderer.cpp:1]: For sure, this will be migrated into the Rendering project.

File: Core/main.cpp (hunk 3 above)
Comment (Lukauigi) [reply Core/main.cpp:11]: Left this until main successfully creates a rendering object which can draw primitives. Duplication will be deleted on next push.

File: Core/main.cpp [hunk 4]
```diff
@@ -6,136 +6,11 @@
 #include <string>
 #include <sstream>
 
-#include "primitives.h"
+//#include "../Rendering/include/Primitives.h"
+//#include "../Rendering/include/Renderer.h"
```
Comment (le-rich) [inline Core/main.cpp:10]: Let's remove commented code from final pushes

File: Core/main.cpp [hunk 5]
```diff
@@ -6,136 +6,11 @@
...
-    GLCall(glDrawElements(shape.drawMode, shape.indices.size(), GL_UNSIGNED_INT, nullptr));
-    GLCall(glBindVertexArray(vao));
-
-    // cleanup
-    GLCall(glDeleteBuffers(1, &vbo));
-    GLCall(glDeleteVertexArrays(1, &vao));
-    GLCall(glDeleteProgram(shader));
-}
+#include <Renderer.h>
+#include <Primitives.h>
```
Comment (le-rich) [inline Core/main.cpp:13]: We should keep files we include that we made inside " " instead of < >

File: Rendering/include/Renderer.h [hunk 6]
```diff
@@ -0,0 +1,39 @@
+#ifndef RENDERER_H
+#define RENDERER_H
+
+#include <glad/glad.h>
+#include <string>
+
+#include <Primitives.h>
+#include <Renderer.h>
```
Comment (le-rich) [inline Rendering/include/Renderer.h:8]: ditto header style

File: Rendering/src/Renderer.cpp [hunk 7]
```diff
@@ -0,0 +1,132 @@
+#include <glad/glad.h>
+
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <sstream>
+
+#include <Primitives.h>
```
Comment (le-rich) [inline Rendering/src/Renderer.cpp:8]: ditto header style

File: Rendering/include/Renderer.h [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+// Macro definitions for error handling
+#define ASSERT(x) if (!(x)) __debugbreak();
+#define GLCall(x) GLClearError();\
+    x;\
+    ASSERT(GLLogCall(#x, __FILE__, __LINE__))
+
+class Renderer
+{
+public:
+    static const std::string vertexShader;
```
Comment (le-rich) [inline Rendering/include/Renderer.h:19]: static const should be styled VERTEX_SHADER

File: Rendering/include/Renderer.h [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+    static const std::string fragmentShader;
```
Comment (le-rich) [inline Rendering/include/Renderer.h:20]: ditto const style

**Review by le-rich — CHANGES_REQUESTED** [review]
Logic and substance looks good, styling changes requested

**Comment by Lukauigi** [comment]
Do I worry about extrapolating the basic/default shaders and maybe other things like: error handling and utilizing Tissure's buffer classes in other branches?

**Comment by Tissure** [comment]
It's fine for now. You have two other tickets for those anyways.

**Review by Tissure — APPROVED** [review]
LGTM. Compiles and runs on my machine.

**Review by le-rich — APPROVED** [review]

**Pull request merged**